pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java

```diff
@@ -74,6 +76,85 @@ public static int arrayIndexOfString(String[] values, String valueToFind) {
     return ArrayUtils.indexOf(values, valueToFind);
   }
 
+  @ScalarFunction
+  public static int[] arrayIndexOfAllInt(int[] value, int valueToFind) {
+    List<Integer> indices = new ArrayList<>();
```

Comment:
Let's use `IntArrayList` from `it.unimi.dsi.fastutil.ints`, then we can use `list.toIntArray()`